common/lib/dependabot/pull_request_creator.rb

```diff
@@ -197,7 +198,7 @@ def codecommit_creator
     end
 
     def message_builder
-      @message_builder ||
+      @message ||
```

Comment:
Pretty sure this should have been memoized with `||=`, when it's not set we'll initialise and build 3 times

Maybe also clearer if we renamed the method:

```
def message
  @message ||=
    MessageBuilder.new...
end
```

Not sure if it's worth it but we could also add a `message` method on the Builder which wraps the three arguments in the `Dependabot::PullRequestCreator::Message` class so we always access things through the same class (interfaces would be neat..)

Reply:
> Not sure if it's worth it but we could also add a message method on the Builder which wraps the three arguments

I'd done that, but I didn't dig amending the `PullRequestCreator::MessageBuilder` tests enough to keep it around. I'll bring it back.

Reply:
> Pretty sure this should have been memoized with ||=, when it's not set we'll initialise and build 3 times
> rename the method

Good shout, done and tested in  https://github.com/dependabot/dependabot-core/pull/3191/commits/b55cbc8f3c46b7e4a1e75163c45f857845c939b8

> add a message method on the Builder 

Added in https://github.com/dependabot/dependabot-core/pull/3191/commits/ea09d928d1ca0760039dd23da1e81b128e358aae , incl the weak test that initially made me question the value.